### DISCUSSION TITLE
Hub-01 Expanded Merchant Item Pool

### DISCUSSION THREAD

**Discussion by Admiral-Daala1541** [cataclysmbn/Cataclysm-BN#3823] [General]
So, I was thinking about how on the discord we recently talked about Hub-01s focus pre-zero hour being on materials research.

Leaning into that, I was wondering about how feasible it could be to make their merchant a renewable source of factory grade 8x40, Hydrogen cannisters, CRIT pulse ammo, and so on. And how we should balance that becoming available.

My first thought was locking its spawn behind the Nanofab quest somehow, but im more than open to ideas other than "make it ridiculously expensive".

**Comment by chaosvolt** [comment]
> My first thought was locking its spawn behind the Nanofab quest somehow

Since The Intercom is the merchant, spawning the desired items in the space next to them in their lil hidey-hole would work, along with making the area they spawn in have a faction ownership so whatever is spawned on completing that mission gets marked as for sale.

This can be used to spawn limited amounts of materials that shouldn't be allowed to restock, as opposed to general shopkeep inventory.

**Reply by Admiral-Daala1541** [reply]
The only issue with that is that the implication I was going off of is that they have a (now working after getting the parts data they needed) nanofab in there, and a steaady supply of resin for it. So in theory the unlocked items *should* be able to restock.

Unless we just wanna give them such a huge stockpile that there's no way a player can buy it all up in one playthrough,, which would look a bit silly even if it would work.

**Reply by chaosvolt** [reply]
> So in theory the unlocked items _should_ be able to restock.

Reasonable yeah, just that code-wise the problems with it are:
1. From looking at the docs it seems we don't actually have a dialogue function for placing items directly into an NPC's inventory without taking it from the player's inventory instead. We have functions for giving items to the player both from NPC inventory and from out of thin air, but don't seem to have a to-NPC-inventory version of the latter.
2. We also don't have a way to explicitly change what items are in a shopkeeper's inventory itemgroup mid-game via dialogue functions, so the only way it'd work in their restocking inventory is if we just said "fuck it" and allowed them to sell Rivtech ammo from day one via adding it directly to the offending itemgroup.

Balance-wise, having some shopkeeper inclusions that unlock with specific missions completed and don't restock might be reasonable as an itemgroup dedicated to more powerful stuff that we want to limit the availability of.

**Reply by Admiral-Daala1541** [reply]
What about adding a second vendor that only unlocks after their existing questline. Is that doable? Since I know we already have the free merchant and his body guards spawning in upstairs, could be a way to reintroduce the basement in having the second vender spawn down there under the assumption that its easier to do business there away from prying eyes. And you get to/talk to him via a subway access connection.

Also, while i understand that for say, mininukes and *maybe* hydrogen (if the difference between homemade/commercial and factory grade stuff wasn't so severe), I don't really see that for 8x40 when 50 bmg exists and is reloadable.

That all being said, one thing that didn't occur to me straight away was the uncraftable carbon fiber and aluminum bolt/arrow types. Which they probably would be willing to sell right off the bat since they're basically the equivalent of banging rocks together for the faction.

**Reply by chaosvolt** [reply]
Yeah that'd work as a good start, allow limited access to the basement and thus a merchant who can be set to stock stuff too spicy for the intercom to sell. This would also allow reintroducing the basement layer piece by piece, start with only the bits nessecary for the player to receive authorization to deal with a merchant downstairs, then reconstruct the basement layer as needed should a reason to explore more of it gets needed.
